### PR TITLE
Fix #264 and add tests.

### LINKED ISSUE #264: ScriptJob broken with data container
Same issue as https://github.com/pyiron/pyiron_base/issues/44

```
from pyiron import Project
pr = Project('JOB')
job = pr.create_job('ScriptJob', 'job')
job.input['value'] = 300
job.script_path = 'test.ipynb'
job.run()
```
in `test.ipynb`: 

```
from pyiron import Notebook as nb
coeff_tot = nb.get_custom_dict()
print(coeff_tot['value'])
```

The error message I get: 

```
ValueError                                Traceback (most recent call last)
<ipython-input-1-34e7bd0a0faf> in <module>
      1 from pyiron import Notebook as nb
----> 2 coeff_tot = nb.get_custom_dict()
      3 print(coeff_tot['value'])

~/PycharmProjects/pyiron_base/pyiron_base/job/external.py in get_custom_dict()
     38         if Path(hdf_file).exists():
     39             hdf = FileHDFio(hdf_file)
---> 40             custom_dict = hdf[folder + '/input/custom_dict/data']
     41             custom_dict["project_dir"] = str(project_folder)
     42             return custom_dict

~/PycharmProjects/pyiron_base/pyiron_base/generic/hdfio.py in __getitem__(self, item)
    833                     hdf_object = self.copy()
    834                     hdf_object.h5_path = "/".join(item_abs_lst[:-1])
--> 835                     return hdf_object[item_abs_lst[-1]]
    836 
    837     def _read(self, item):

~/PycharmProjects/pyiron_base/pyiron_base/generic/hdfio.py in __getitem__(self, item)
    805                         obj = hdf_item.copy()
    806                         return obj
--> 807                 raise ValueError("Unknown item: {}".format(item))
    808             else:
    809                 if item_lst[0] == "":  # item starting with '/', thus we have an absoute HDF5 path

ValueError: Unknown item: data
```

### PR DESCRIPTION
**Pull request by pmrv**
Seems you were a bit quicker, but from my test #47 still works fine and has the tests already.

### CODE REVIEW

File: pyiron_base/job/external.py
Comment (jan-janssen) [inline pyiron_base/job/external.py]: I guess using an internal function is not recommended. Maybe we should make it a public function.

Comment (pmrv) [reply pyiron_base/job/external.py]: The same method is also used by `GenericJob`, so making it public is probably ok.

Comment (pmrv) [reply pyiron_base/job/external.py]: See #268.

**Review by jan-janssen — APPROVED** [review]
The code looks good to me - so once the tests pass feel free to merge it. 

Ps: in addition we should run the integration tests once as maybe somebody - e.g. me - was using the private function in `pyiron_atomistics`.

**Comment by jan-janssen** [comment]
@pmrv I am a bit pushing on this pull request as we have users on the cluster who need this rather quickly.

**Comment by pmrv** [comment]
> The code looks good to me - so once the tests pass feel free to merge it.
> 
> Ps: in addition we should run the integration tests once as maybe somebody - e.g. me - was using the private function in `pyiron_atomistics`.

I grepped through atomistics and contrib but there doesn't seem to be any calls of `_create_project_from_hdf5` in either.

**Comment by pmrv** [comment]
@niklassiemer I guess neither me nor @jan-janssen have a windows installation ready to go, can you have a look what's going on with the windows paths?

**Comment by niklassiemer** [comment]
Yes, I will have a look this evening - probably starting around 8pm.

**Comment by pmrv** [comment]
> Yes, I will have a look this evening - probably starting around 8pm.

Thanks a lot!

**Comment by jan-janssen** [comment]
Finally I used the solution proposed in https://github.com/pyiron/pyiron_base/pull/266

Looking at the previous implementation in detail, we create a FileHDFio object, from it we create a Project, from the project we open the HDF5 to get a ProjectHDF object and then we call to_object() to get a DataContainer. In the new implementation I go from a FileHDFio object directly to DataContainer.

File: tests/job/test_external.py
Comment (liamhuber) [inline tests/job/test_external.py:12]: While we're touching this it would be very nice to convert it over to subclass of `pyiron_base._tests.TestWithProject` and remove a bunch of the setup and teardown code.

Comment (jan-janssen) [reply tests/job/test_external.py:12]: As we need the changes in the pull request rather urgently I would move that to a separate pull request

File: tests/job/test_scriptjob.py
Comment (liamhuber) [inline tests/job/test_scriptjob.py:6]: Similarly here. 

I don't object if anyone wants to go through and implement the new parentage everywhere it applies, but IMO it's sufficient if we just update things as we come across them.

File: tests/job/test_external.py
Comment (liamhuber) [reply tests/job/test_external.py:12]: It's five minutes. How urgent is urgent? Let me put my kid to bed and I'll do it (~20:15).

Comment (jan-janssen) [reply tests/job/test_external.py:12]: I proposed the fix 4 hours ago https://github.com/pyiron/pyiron_base/pull/266  and we still need to create the conda package plus wait for the release to be available on the cluster. I guess the update of the test can also be done in a separate pull request and it is not required on the cluster.

**Comment by pmrv** [comment]
> Finally I used the solution proposed in #266
> 
> Looking at the previous implementation in detail, we create a FileHDFio object, from it we create a Project, from the project we open the HDF5 to get a ProjectHDF object and then we call to_object() to get a DataContainer. In the new implementation I go from a FileHDFio object directly to DataContainer.

Agree that our Project/HDF layer has too many indirections.

**Comment by jan-janssen** [comment]
@muh-hassani Can you update `pyiron_base` to 0.2.8 to fix the ScriptJob issue?

**Pull request merged**
